src/http/client.cc

```diff
@@ -346,14 +350,14 @@ future<> client::make_request(request req, reply_handler handle, std::optional<r
             // The 'con' connection may not yet be freed, so the total connection
             // count still account for it and with_new_connection() may temporarily
             // break the limit. That's OK, the 'con' will be closed really soon
-            return with_new_connection([this, &req, &handle, expected] (connection& con) {
-                return do_make_request(con, req, handle, expected);
-            });
+            return with_new_connection([this, &req, &handle, as, expected] (connection& con) {
+                return do_make_request(con, req, handle, as, expected);
+            }, as);
         });
     });
 }
 
-future<> client::do_make_request(connection& con, request& req, reply_handler& handle, std::optional<reply::status_type> expected) {
+future<> client::do_make_request(connection& con, request& req, reply_handler& handle, abort_source* as, std::optional<reply::status_type> expected) {
     return con.do_make_request(req).then([&con, &handle, expected] (connection::reply_ptr reply) mutable {
```

Comment:
Does it not make sense to share the abort_source in the client class?

Why does each request need a separate abort_source?

Reply:
Because the intent is to abort individual make_request, not the whole client. Scylla uses [single client for configured S3 endpoint](https://github.com/scylladb/scylladb/pull/13859). If there's a [keyspace with storage=s3](https://github.com/scylladb/scylladb/pull/10144) and [backup activity](https://github.com/scylladb/scylladb/pull/19890) towards the same endpoint, if user wants to abort only backup, it should abort only some subset of in-flight requests, not the whole client.

Reply:
Why not create a client per {set of related requests}?

Reply:
Then we'll need to implement resource control in a different manner. Resources in the scope of s3 client are -- memory buffers (we flush sstables in 5Mb parts) and the number of sockets in client pool. Or leave different {sets of related requests} consume resources independently

Reply:
@avikivity , reminder ping

Reply:
Sorry. Could we extract the resource controller out? So multiple clients can share the same resource pool?

Though it's not out of the question to want to abort a particular request.

Reply:
> Sorry. Could we extract the resource controller out? So multiple clients can share the same resource pool?

Yes, it's possible. However, http::client is all about socket pool management, so sharing sockets pool is the same as sharing the http client itself.

> Though it's not out of the question to want to abort a particular request.

Yes. Also, putting abort source on client will not change much in the code, just instead of abort_source& argument there will be this->abort_source